src/services/assetService.ts

```diff
@@ -129,8 +129,9 @@ export class AssetService {
             if (!normalizedPath.startsWith("http://") &&
                 !normalizedPath.startsWith("https://") &&
                 !normalizedPath.startsWith("file:")) {
-                asset.path = "file:" + asset.path;
-                asset.path = encodeURI(asset.path.replace(/\\/g, "/"));
+                let path = encodeURI("file:" + asset.path.replace(/\\/g, "/"));
+                path = path.replace(/#/g, "%23");
+                asset.path = path;
```

Comment:
Instead of 1 off special case solution I would rather we refactor this to use `encodeURIComponent(...)` method to encode the filename portion of the path.  The fix may need to be inside of `AssetService.createAssetFromFilePath`

https://stackoverflow.com/questions/4540753/should-i-use-encodeuri-or-encodeuricomponent-for-encoding-urls

I would also like to see some unit tests for these bug fixes.

Reply:
Actually I see encodeURIComponent will encode everything, slashes included.  It seams to be an api recomended for encoding query strings and not entire URI.
`var uri = 'http://example.com/foo?hello=' + encodeURIComponent(world);`

The problem with encodeURI, that instead it assumes that the input is a complete URI, is that it doesn't encode ~!@#$&*()=:/,;?+'

I found this escamotage for encodeURI here: https://stackoverflow.com/questions/6627648/how-can-i-submit-a-hash-key-in-an-url-parameter

Anyway I think best method to call in our case should be **escape()**

Reply:
escape is encoding unfortunately the : character you may have on windows path

Reply:
I'm implementing a custom encoding table and generilized regular expression to handle all exception cases from encodeURI() that we want to handle.